pkg/db/db_sync_status.go

```diff
@@ -0,0 +1,244 @@
+		syncData.SyncStatus)
+
+	if err != nil {
+		return fmt.Errorf("could not write sync event into DB. Error: %w\n", err)
+	}
+	return nil
+}
+
+func (h *DBHandler) DBWriteNewSyncEventBulk(ctx context.Context, tx *sql.Tx, id TransformerID, envApps []EnvApp, status SyncStatus) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBWriteNewSyncEventBulk")
+	defer span.Finish()
+	if h == nil {
+		return nil
+	}
+	if tx == nil {
+		return fmt.Errorf("DBWriteNewSyncEventBulk: no transaction provided")
+	}
+	now, err := h.DBReadTransactionTimestamp(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("DBWriteNewSyncEventBulk unable to get transaction timestamp: %w", err)
+	}
+
+	err = h.executeBulkInsert(ctx, tx, envApps, *now, id, status, BULK_INSERT_BATCH_SIZE)
+
+	if err != nil {
+		return fmt.Errorf("could not write sync event into DB. Error: %w\n", err)
+	}
+	return nil
+}
+
+type EnvApp struct {
+	AppName string
+	EnvName string
+}
+
+func (h *DBHandler) DBReadUnsyncedAppsForTransfomerID(ctx context.Context, tx *sql.Tx, id TransformerID) ([]EnvApp, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBReadUnsyncedAppsForTransfomerID")
+	defer span.Finish()
+	if h == nil {
+		return nil, nil
+	}
+	if tx == nil {
+		return nil, fmt.Errorf("DBReadUnsyncedAppsForTransfomerID: no transaction provided")
+	}
+	selectQuery := h.AdaptQuery("SELECT appName, envName FROM git_sync_status WHERE transformerid = ? AND status = ? ORDER BY eslVersion;")
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuery,
+		id,
+		UNSYNCED,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not get current eslVersion. Error: %w\n", err)
+	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("row closing error: %v", err)
+		}
+	}(rows)
+	allCombinations := make([]EnvApp, 0)
+	var currApp string
+	var currEnv string
+	for rows.Next() {
+		err := rows.Scan(&currApp, &currEnv)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("Error table for next eslVersion. Error: %w\n", err)
+		}
+		allCombinations = append(allCombinations, EnvApp{
+			AppName: currApp,
+			EnvName: currEnv,
+		})
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return allCombinations, nil
+}
+
+func (h *DBHandler) DBBulkUpdateUnsyncedApps(ctx context.Context, tx *sql.Tx, id TransformerID, status SyncStatus) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBBulkUpdateUnsyncedApps")
+	defer span.Finish()
+	if h == nil {
+		return nil
+	}
+	if tx == nil {
+		return fmt.Errorf("DBBulkUpdateUnsyncedApps: no transaction provided")
+	}
+
+	allCombs, err := h.DBReadUnsyncedAppsForTransfomerID(ctx, tx, id)
+	if err != nil {
+		return fmt.Errorf("DBBulkUpdateUnsyncedApps unable to read unsynced apps: %w", err)
+	}
+	if len(allCombs) == 0 {
+		logger.FromContext(ctx).Sugar().Warnf("Could not update all unsynced apps. Did not find any unsynced apps.")
```

Comment:
that's not really a problem, is it?
Either remove the warning or turn it into "info".